atom/browser/ui/message_box_win.cc

```diff
@@ -183,98 +187,57 @@ int ShowTaskDialogUTF16(NativeWindow* parent,
     return cancel_id;
 }
 
-int ShowTaskDialogUTF8(NativeWindow* parent,
-                       MessageBoxType type,
-                       const std::vector<std::string>& buttons,
-                       int default_id,
-                       int cancel_id,
-                       int options,
-                       const std::string& title,
-                       const std::string& message,
-                       const std::string& detail,
-                       const std::string& checkbox_label,
-                       bool* checkbox_checked,
-                       const gfx::ImageSkia& icon) {
+int ShowTaskDialogUTF8(const MessageBoxSettings& settings) {
   std::vector<base::string16> utf16_buttons;
-  for (const auto& button : buttons)
+  for (const auto& button : settings.buttons)
     utf16_buttons.push_back(base::UTF8ToUTF16(button));
 
+  base::string16 title_16 = base::UTF8ToUTF16(settings.title);
+  base::string16 message_16 = base::UTF8ToUTF16(settings.message);
+  base::string16 detail_16 = base::UTF8ToUTF16(settings.detail);
+  base::string16 checkbox_label_16 = base::UTF8ToUTF16(settings.checkbox_label);
+  bool checkbox_checked16 = settings.checkbox_checked;
```

Comment:
_(nit)_ all the above can be `const`

_(nit)_ the variable name "checkbox_checked16" is fun. Converting a bool to a 16-bit bool is an interesting choice :neutral_face: